Cargo.lock

```diff
@@ -282,7 +322,7 @@ checksum = "db8b7511298d5b7784b40b092d9e9dcd3a627a5707e4b5e507931ab0d44eeebf"
 dependencies = [
  "proc-macro2",
  "quote",
- "syn",
+ "syn 1.0.109",
```

Comment:
Based on these changes, I assume you ran `cargo update --all`?

I think you should revert that and only update the frontier deps (e.g. `cargo update -p fp-evm` will do only the frontier ones).

If kept like this, we introduce changes to potentially everything, not just `frontier` related stuff.

Reply:
Is keep updated is a problem?

Reply:
Imo, it's a matter of what we update & when.

E.g. the part I highlighted introduces a new version of an already used dependency, so we end up having two versions of `syn` in the build. 

Usually we bump versions as we bump `polkadot` revisions automatically - that way we can at least be sure that the deps we use are the same ones as used by `polkadot` and they do seem to constantly audit their PRs and releases.We can have some confidence in version bumps introduced by them, while if we do it like this is more like an assumption that it should all be fine.

And the purpose of this PR isn't to update everything, it's only to bring in latest version of `frontier` with the included fix, right?
Keeping it localized and changes to the minimum is therefore preferred.

And also, let's keep in mind that this is going on `Astar` next week, and we won't have the chance to test these upgrades on `Shiden`.